client/auth.go

```diff
@@ -77,21 +77,25 @@ func (c *Conn) readInitialHandshake() error {
 		c.capability = uint32(binary.LittleEndian.Uint16(data[pos:pos+2]))<<16 | c.capability
 		pos += 2
 
-		// skip auth data len or [00]
+		// auth_data is end with 0x00, max data length is 13 + 8 = 21
+		// ref to https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::Handshake
+		maxAuthDataLen := 21
+		if c.capability&CLIENT_PLUGIN_AUTH != 0 && int(data[pos]) < maxAuthDataLen {
```

Comment:
if `int(data[pos])` is `9`, `resetOfAuthDataEndPos` will equals `pos`. But as stated `$len=MAX(13, length of auth-plugin-data - 8)` in https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::Handshake, `resetOfAuthDataEndPos` should be `pos + 13 - 1`?

Reply:
sorry, it's my fault. it's `min_auth_data_len` is 21, not `max_data_len`.

Reply:
and this pr is a little bit old, need to be updated :-)